File: Doctor.php
<?php

require 'conn.php';

if ($_SERVER['REQUEST_METHOD'] == 'POST') {
    $input = json_decode(file_get_contents('php://input'), true);
    $sql = 'INSERT INTO doctors (DFirstName, DLastName, gender, DSpecialization, address, birthdate, DContactNum) VALUES (?, ?, ?, ?, ?, ?, ?)';

    $stmt = $pdo->prepare($sql);
    $stmt->execute([$input['DFirstName'], $input['DLastName'], $input['gender'], $input['DSpecialization'], $input['address'], $input['birthdate'], $input['DContactNum']]);
    echo json_encode('Doctor added');
}

if ($_SERVER['REQUEST_METHOD'] == 'GET') {
    $sql = 'SELECT DoctorID, DFirstName, DLastName, gender, DSpecialization, address, birthdate, DContactNum FROM doctors';
    $stmt = $pdo->query($sql);
    $doctors = $stmt->fetchAll();
    echo json_encode($doctors);
}

File: Doctor.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using Newtonsoft.Json;

namespace clinicinfosys
{
    public partial class Doctor : Form
    {
        private static readonly HttpClient client = new HttpClient();

        public Doctor()
        {
            InitializeComponent();
            this.Load += LoadDoctors;
        }

        private async void LoadDoctors(object sender, EventArgs e)
        {
            try
            {
                HttpResponseMessage response = await client.GetAsync(
                    "http://localhost/jenApi/Doctor.php"
                );
                response.EnsureSuccessStatusCode();

                string jsonResponse = await response.Content.ReadAsStringAsync();
                List<Doctor2> doctors = JsonConvert.DeserializeObject<List<Doctor2>>(jsonResponse);

                doctorsView.DataSource = doctors;
            }
            catch (Exception ex)
            {
                MessageBox.Show("Error: " + ex.Message);
            }
        }
    }

    public class Doctor2
    {
        public int DoctorID { get; set; }
        public string DFirstName { get; set; }
        public string DLastName { get; set; }
        public string gender { get; set; }
        public string DSpecialization { get; set; }
        public string address { get; set; }
        public string birthdate { get; set; }
        public string DContactNum { get; set; }
    }
}
